cachi2/core/package_managers/metayarn.py

```diff
@@ -0,0 +1,21 @@
+from cachi2.core.errors import PackageRejected
+from cachi2.core.models.input import Request
+from cachi2.core.models.output import RequestOutput
+from cachi2.core.package_managers import yarn, yarn_classic
+from cachi2.core.resolver import _merge_outputs
+
+
+def fetch_yarn_source(request: Request) -> RequestOutput:
+    """Fetch yarn source."""
+    # Packages could be a mixture of yarn v1 and v2 (at least this is how it
+    # looks now). To preserve this behavior each reqiest is split into individual
+    # packages which are assessed one by one.
+    fetched_packages = []
+    for package in request.packages:
+        new_request = request.model_copy(update={"packages": [package]})
+        try:
+            fetched_packages.append(yarn_classic.fetch_yarn_source(new_request))
+        except PackageRejected:
```

Comment:
Good point, thank you. This PR adds two checks: one for lock file presence and another for lock file version. Both raise PackageRejected error to fail early and to signal that a package cannot be processed. This could be problematic if something else raises PackageRejected when processing actual v1 package (for some other reason). I'll subclass PackageRejected with a couple more specific exceptions (one for missing lock file and another for not a v1 lock file) and will catch them here

Reply:
Maybe it'd better to have these new errors as part of either the `metayarn.py` or the `yarn_classic/main.py` module. I feel like errors under `core/errors.py` are meant to be globally used in the application, and we should limit the number of variants there.

Reply:
Good point, thank you. I had doubts about `core/errors.py` being the right home for these exceptions, but placed them there anyway. Will move them in the next push.